### PR TITLE
Update libraries to latest

### PR DESCRIPTION
**Pull request by kenyee**
## 🚀 Description
Update libs to latest
Kotlin to 1.5.21 (I think Jetpack Compose 1.0 will use this)
Gradle to 7.1.1
Target SDK to 31
Removes jcenter which is deprecated

## 📄 Motivation and Context
Template libs are a bit stale

## 🧪 How Has This Been Tested?
Ran test app

## 📷 Screenshots (if appropriate)
N.A.

## 📦 Types of changes
Not bug/feature/breaking.  Just libs update

## ✅ Checklist
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
No doc update needed

### CODE REVIEW

File: .idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -0,0 +1,139 @@
+<component name="ProjectCodeStyleConfiguration">
```
Comment (cortinico) [inline .idea/codeStyles/Project.xml:1]: Where have you generated this config from?

**Review by cortinico — APPROVED** [review]
Thank you very much for the contrib @kenyee 🙏

File: .idea/codeStyles/Project.xml (hunk 1 above)
Comment (kenyee) [reply .idea/codeStyles/Project.xml:1]: this was generated by AS after applying the Kotlin style to the project and deleting java.util.* from the wildcard imports and setting the wildcard import point to 99 entries (we've had problems with that being flagged by Detekt)

**Comment by kenyee** [comment]
not sure why the pre-merge check fails 🤔

File: buildSrc/build.gradle.kts [hunk 2]
```diff
@@ -2,21 +2,20 @@ plugins {
     `kotlin-dsl`
 }
 repositories {
-    jcenter()
```
Comment (cortinico) [inline buildSrc/build.gradle.kts:5]: This should be replaced with `mavenCentral()` (that's also the reason why the CI is failing 👍 ).

Comment (kenyee) [reply buildSrc/build.gradle.kts:5]: hah..nice one.  Totally missed that it was missing because I always thought it was in there by default 😞

Comment (kenyee) [reply buildSrc/build.gradle.kts:5]: weird that builds worked fine for me w/o that though 🤔

**Review by cortinico — APPROVED** [review]
Sorry for the late approval 🙏 Great work 🚀

**Comment by kenyee** [comment]
weird looking build errors on ubuntu 🤔 
Why would ktlint fail on that but not the others?

File: buildSrc/build.gradle.kts [hunk 3]
```diff
@@ -2,21 +2,21 @@ plugins {
...
     google()
+    mavenCentral()
 }
 
 kotlinDslPluginOptions.experimentalWarning.set(false)
 
 object Plugins {
-    const val AGP = "4.1.3"
+    const val AGP = "4.2.2"
     const val DOKKA = "1.4.20"
```
Comment (cortinico) [inline buildSrc/build.gradle.kts:13]: Can you bump this to `1.5.0`?

The CI is failing as the current version is relying on a older version of `kotlinx-html` that lives only on jcenter

**Comment by kenyee** [comment]
weird... updated to Dokka 1.5.0 and AGP 7.0.0 and it gets a different weird error like it can't figure out the dependency for AGP and reads it as null.
I did a project clean/rebuild on MacOS and Windows w/o problems 🤔

**Comment by cortinico** [comment]
> weird... updated to Dokka 1.5.0 and AGP 7.0.0 and it gets a different weird error like it can't figure out the dependency for AGP and reads it as null.
> I did a project clean/rebuild on MacOS and Windows w/o problems 🤔

Can you update the `pre-merge` workflow to use Java 11 instead of Java 8 (that's a requirement of AGP 7.x).

**Comment by cortinico** [comment]
The approach in 342d5b3 is unlikely to work. You need to add `setup-java` to the workflows just after the checkouts: https://github.com/actions/setup-java

Like
```
- uses: actions/setup-java@v2
  with:
    distribution: 'adopt'
    java-version: '11'
```

**Comment by kenyee** [comment]
sorry....dug a bit more and confirmed that should work.  Crossing fingers...
I wish there a way just to ask for a java11 agent...

**Comment by kenyee** [comment]
Well..that looks like it failed too :-(
Is there any command line Github Actions YAML validator?  All I could find was a github action to validate the YAML which makes no sense 🤔

**Comment by cortinico** [comment]
> Is there any command line Github Actions YAML validator? All I could find was a github action to validate the YAML which makes no sense 🤔

No problem :) I've pushed a small fix to make it work.
Thank you for the overall work 🙏

**Comment by kenyee** [comment]
> No problem :) I've pushed a small fix to make it work.

Ahh...now it makes sense.  It's a separate build step.
But doesn't this also mean the other places where I added "use" aren't needed?  I thought it was similar to asking for some java config for that build step.

**Comment by cortinico** [comment]
> But doesn't this also mean the other places where I added "use" aren't needed?

I've cleaned them all 👍

**Pull request merged**